src/services/priceService.spec.ts

```diff
@@ -1,18 +1,144 @@
 /* eslint-disable jest/expect-expect, jest/no-disabled-tests */
 
-describe.skip('Price Service', () => {
+import { createMockInstance } from '@common/utils/tests';
+import axios, { AxiosInstance } from 'axios';
+import { toToken } from '@common/utils/currency';
+import { parseUnits } from '@ethersproject/units';
+import { DEFILLAMA_IDS, STABLE_COINS } from '@constants';
+
+import WalletService from './walletService';
+import ContractService from './contractService';
+import ProviderService from './providerService';
+import SdkService from './sdkService';
+import PriceService from './priceService';
+
+jest.mock('./walletService');
+jest.mock('./contractService');
+jest.mock('./providerService');
+jest.mock('./sdkService');
+jest.mock('axios');
+
+const MockedSdkService = jest.mocked(SdkService, { shallow: false });
+const MockedWalletService = jest.mocked(WalletService, { shallow: true });
+const MockedContractService = jest.mocked(ContractService, { shallow: true });
+const MockedProviderService = jest.mocked(ProviderService, { shallow: true });
+
+describe('Price Service', () => {
+  let walletService: jest.MockedObject<WalletService>;
+  let contractService: jest.MockedObject<ContractService>;
+  let providerService: jest.MockedObject<ProviderService>;
+  let sdkService: jest.MockedObject<SdkService>;
+  let priceService: PriceService;
+
+  const mockGetHistoricalPricesForChain = jest.fn();
+  const mockGetCurrentPricesForChain = jest.fn();
+  let getTokenAddressForPriceFetchingSpy: jest.SpyInstance;
+
+  const tokenAAddress = 'token-a';
+  const tokenBAddress = 'token-b';
+  const tokenCAddress = 'token-c';
+
+  beforeEach(() => {
+    walletService = createMockInstance(MockedWalletService);
+    contractService = createMockInstance(MockedContractService);
+    providerService = createMockInstance(MockedProviderService);
+    sdkService = createMockInstance(MockedSdkService);
+    sdkService.sdk = {
+      ...sdkService.sdk,
+      priceService: {
+        getHistoricalPricesForChain: mockGetHistoricalPricesForChain,
+        supportedChains: jest.fn(),
+        supportedQueries: jest.fn(),
+        getCurrentPricesForChain: mockGetCurrentPricesForChain,
+        getCurrentPrices: jest.fn(),
+        getHistoricalPrices: jest.fn(),
+      },
+    };
+    priceService = new PriceService(
+      walletService,
+      contractService,
+      axios as AxiosInstance,
+      providerService,
+      sdkService
+    );
+
+    // eslint-disable-next-line global-require, @typescript-eslint/no-var-requires
+    getTokenAddressForPriceFetchingSpy = jest.spyOn(require('@constants/addresses'), 'getTokenAddressForPriceFetching');
```

Comment:
I initially gave the `jest.requireActual` approach a shot, similar to what's in `walletService.spec.ts`. It's neat for mocking parts of a module. However, I leaned towards `jest.spyOn()` because I wanted the mock to only affect one specific test case. With `jest.requireActual`, I found myself having to specify the mocked value for `getTokenAddressForPriceFetching` for each test